assembly/src/assembler/mod.rs

```diff
@@ -310,6 +341,67 @@ impl Assembler {
 
 /// Compilation/Assembly
 impl Assembler {
+    /// Assembles a set of modules into a library.
+    ///
+    /// The returned library can be added to the assembler assembling a program that depends on the
+    /// library using [`Self::add_compiled_library`].
+    pub fn assemble_library(
+        mut self,
+        modules: impl Iterator<Item = impl Compile>,
+        metadata: CompiledLibraryMetadata, // name, version etc.
+    ) -> Result<CompiledLibrary, Report> {
+        let module_ids: Vec<ModuleIndex> = modules
+            .map(|module| {
+                let module = module.compile_with_options(CompileOptions::for_library())?;
+
+                if module.path().namespace() != &metadata.name {
```

Comment:
This can't be a restriction imposed on libraries - as they can include things from multiple MASM namespaces.

Reply:
Library metadata was removed for now